### PR TITLE
Fix the make test-coverage for s390x arch

### LINKED ISSUE #3756: 'make test-coverage' failed on s390x platform
/kind bug

## What versions of software are you using?

**Operating System:**
Red Hat Enterprise Linux 8.2 (Ootpa)
**Output of `odo version`:**
odo v1.2.5 (0246cf66c)
Server: https://api.ocp43b.rhocp.boe:6443
Kubernetes: v1.18.3+012b3ec
## How did you run odo exactly?
locally
## Actual behavior
make test-coverage

## Expected behavior
passed

## Any logs, error output, etc?
./scripts/generate-coverage.sh
go test: -race is only supported on linux/amd64, linux/ppc64le, linux/arm64, freebsd/amd64, netbsd/amd64, darwin/amd64 and windows/amd64
make: *** [Makefile:115: test-coverage] Error 2

### PR DESCRIPTION
**Pull request by zhengxiaomei123**
**What type of PR is this?**
 /kind bug

**What does does this PR do / why we need it**:
support `make test-coverage` for s390x arch
**Which issue(s) this PR fixes**:

Fixes #3756 

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: scripts/generate-coverage.sh [hunk 1]
```diff
@@ -4,12 +4,20 @@
 # go test can't generate code coverage for multiple packages in one command
 
 set -e
+ARCH=$(uname -m)
 echo "" > coverage.txt
-go test -i -race ./cmd/odo
+if [ "${ARCH}" == "s390x" ]; then
+    go test -i ./cmd/odo
```
Comment (prietyc123) [inline scripts/generate-coverage.sh:10]: please add the comment why you are disabling `-race` flag here. As I can see in [comment](https://github.com/openshift/odo/issues/3756#issuecomment-674852273) its not supported so would suggest you to update it here as well

File: scripts/generate-coverage.sh [hunk 2]
```diff
@@ -4,12 +4,20 @@
...
+else
+    go test -i -race ./cmd/odo
+fi
+
 for d in $(go list ./... | grep -v vendor | grep -v tests); do
     # For watch related tests, race check causes issue so disabling them here as race is already tested in other tests when used with `-coverprofile=profile.out`
     if [ "$d" = "github.com/openshift/odo/pkg/component" ]; then
         go test -coverprofile=profile.out -covermode=atomic $d
+    elif [ "${ARCH}" == "s390x" ]; then
```
Comment (prietyc123) [inline scripts/generate-coverage.sh:19]: https://github.com/openshift/odo/pull/3765#discussion_r472060791

Comment (zhengxiaomei123) [reply scripts/generate-coverage.sh:20]: @prietyc123 Done. Thanks.

File: scripts/generate-coverage.sh (hunk 1 above)
Comment (zhengxiaomei123) [reply scripts/generate-coverage.sh:11]: @prietyc123 Done. Thanks.

**Comment by prietyc123** [comment]
looks good to me

/lgtm

**Pull request merged**